pkg/controller/elasticsearch/client/client_test.go

```diff
@@ -277,6 +277,16 @@ func TestAPIError_Error(t *testing.T) {
 				`Type:illegal_argument_exception} Reason:illegal value can't update [discovery.zen.minimum_master_nodes] from [1] to [6] Type:illegal_argument_exception ` +
 				`StackTrace: RootCause:[{Reason:[stack-sample-es-575vhzs8ln][10.60.1.22:9300][cluster:admin/settings/update] Type:remote_transport_exception}]}}`,
 		},
+		{
```

Comment:
nit: the changes in `pkg/controller/elasticsearch/client/error.go` have no effect on the value returned by `apiError.Error()`, unless I'm missing something I don't think this test brings more value than the existing ones?

Reply:
You are right about that. I added it because I felt we were missing the case where we have an HTTP error status code but a regular Elasticsearch REST response in the body of the request.